assets/js/modules/tagmanager/components/setup/SetupForm.js

```diff
@@ -16,64 +16,105 @@
+	const submitMode = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_SETUP, 'submitMode' ) );
+	const analyticsModuleReauthURL = useSelect( ( select ) => select( MODULES_ANALYTICS ).getAdminReauthURL() );
 
 	const { setValues } = useDispatch( CORE_FORMS );
-	const { submitChanges } = useDispatch( STORE_NAME );
-	const submitForm = useCallback( async ( event ) => {
-		event.preventDefault();
+	const { activateModule } = useDispatch( CORE_MODULES );
+	const { submitChanges, receiveError } = useDispatch( STORE_NAME );
+	const dispatchAnalytics = useDispatch( MODULES_ANALYTICS );
+	const submitForm = useCallback( async ( { submitMode } ) => { // eslint-disable-line no-shadow
+		// We'll use form state to persist the chosen submit choice
+		// in order to preserve support for auto-submit.
+		setValues( FORM_SETUP, { submitMode } );
+
 		const { error } = await submitChanges();
+
 		if ( isPermissionScopeError( error ) ) {
 			setValues( FORM_SETUP, { autoSubmit: true } );
-		}
-		if ( ! error ) {
+		} else if ( ! error ) {
 			setValues( FORM_SETUP, { autoSubmit: false } );
+			// If the property ID is set in GTM and Analytics is active,
+			// we disable the snippet output via Analyics to prevent duplicate measurement.
+			if ( gtmAnalyticsPropertyID && analyticsModuleActive ) {
+				dispatchAnalytics.setUseSnippet( false );
+				const saveAnalyticsSettings = await dispatchAnalytics.saveSettings();
+				if ( saveAnalyticsSettings.error ) {
+					receiveError( saveAnalyticsSettings.error );
+					return;
+				}
+			}
+			// If submitting with Analytics setup, and Analytics is not active,
+			// activate it, and navigate to its reauth/setup URL to proceed with its setup.
+			if ( submitMode === 'with_analytics_setup' && ! analyticsModuleActive ) {
+				setIsNavigating( true );
+				const activateAnalytics = await activateModule( 'analytics' );
+				if ( activateAnalytics.error ) {
+					receiveError( activateAnalytics.error );
+					return;
+				}
+				global.location.assign( analyticsModuleReauthURL );
+				// Don't call finishSetup as this navigates to a different location.
+				return;
+			}
 			finishSetup();
 		}
-	}, [ canSubmitChanges, finishSetup ] );
+	}, [ finishSetup, dispatchAnalytics, gtmAnalyticsPropertyID, analyticsModuleActive, analyticsModuleReauthURL, receiveError ] );
 
 	// If the user lands back on this component with autoSubmit and the edit scope,
 	// resubmit the form.
 	useEffect( () => {
 		if ( autoSubmit && hasEditScope ) {
-			submitForm( { preventDefault: () => {} } );
+			submitForm( { submitMode } );
 		}
-	}, [ hasEditScope, autoSubmit, submitForm ] );
+	}, [ hasEditScope, autoSubmit, submitForm, submitMode ] );
+
+	// Form submit behavior now varies based on which button is clicked,
+	// so we no longer support the generic form submit event.
+	const preventDefaultFormSubmit = useCallback( ( event ) => event.preventDefault(), [] );
```

Comment:
Agreed, we should be able to use the regular form submit event and detect the current `submitMode` from the form? Maybe we could have the two submit buttons with `name` attributes? This way only the one clicked should actually be sent, if I'm not mistaken.

Reply:
@felixarntz that would work in a traditional form submission where the form submit is triggering the navigation passing through its values in GET/POST. In this case, we need the choice to be set in the datastore before continuing.
It's true that we have access to the submitting `HTMLElement` in the form submit event so maybe we can use that. It seems somewhat less correct in terms of React to use the DOM for this though, so I'll see what we can do.

@tofumatt – regarding your point about the AC, it does explicitly say that "Continue to Analytics setup" should be the primary action in this case, so if we're going to change this that's okay with me, but just want to confirm this with @felixarntz.

I'll work on restoring the form submit handler rather than relying on the `onClick` handlers, and if we want to swap which button is primary/secondary for some case, that shouldn't be a big deal to change.